server/schedule/placement/rule_list.go

```diff
@@ -203,5 +203,8 @@ func (rl ruleList) getRulesForApplyRegion(start, end []byte) []*Rule {
 	if i != len(rl.ranges) && (len(end) == 0 || bytes.Compare(end, rl.ranges[i].startKey) > 0) {
 		return nil
 	}
+	if i == 0 {
```

Comment:
when will it happen?

Reply:
The `if` statement is unnecessary here since a default rule is inserted during initialization. I added the initialization code in unit tests and removed the statement.